cmd/secrets/backend_selector.go

```diff
@@ -0,0 +1,57 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build secrets
+
+package secrets
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"time"
+)
+
+type backendPrefix string
+
+const (
+	file       backendPrefix = "file"
+	kubeSecret backendPrefix = "kube_secret"
+)
+
+func (prefix backendPrefix) isValid() bool {
+	return prefix == file || prefix == kubeSecret
+}
```

Comment:
This is more of an architectural structuring thing but as I read this more, I think there's definitely enough `backend` code to move it into its own folder (e.g. `./backend` / `./providers` / etc).

I think what might be easiest to maintain and write tests for would be something like:
```
- cmd/secrets/providers/providers.go (contains interface, their list/map, and public helpers like the selection and "isValid")
- cmd/secrets/providers/file.go (fileBackend impl)
- cmd/secrets/providers/k8s_secret.go or cmd/secrets/providers/k8s.go or cmd/secrets/providers/k8s/secret.go (k8s_secret impl)
```

Reply:
I thought about this while coding, but I didn't consider it necessary.
After your suggestion, I tried it, and I think it makes the code a bit cleaner. I've changed it :+1:

I'll probably need to move some of the existing tests in `secret_helper_test.go` to the `FileProvider` unit tests.